src/node/txreconciliation.cpp

```diff
@@ -116,24 +115,11 @@ class TxReconciliationTracker::Impl
         // v1 is the lowest version, so suggesting something below must be a protocol violation.
         if (recon_version < 1) return PROTOCOL_VIOLATION;
 
-        // Must match SENDTXRCNCL logic.
-        const bool they_initiate = is_peer_recon_initiator && is_peer_inbound;
-        const bool we_initiate = !is_peer_inbound && is_peer_recon_responder;
-
-        // If we ever announce support for both requesting and responding, this will need
-        // tie-breaking. For now, this is mutually exclusive because both are based on the
-        // inbound flag.
-        assert(!(they_initiate && we_initiate));
-
-        // The peer set both flags to false, we treat it as a protocol violation.
-        if (!(they_initiate || we_initiate)) return PROTOCOL_VIOLATION;
-
-        LogPrintLevel(BCLog::TXRECONCILIATION, BCLog::Level::Debug, "Register peer=%d with the following params: " /* Continued */
-                                                                    "we_initiate=%i, they_initiate=%i.\n",
-                      peer_id, we_initiate, they_initiate);
```

Comment:
I think you should update `SENDTXRCNCL` , this still mentioning the role assignment based on booleans. I would also suggest to document somewhere, in `protocol.h` or `txreconciliation.cpp`, near to `m_we_initiate` how the role assignment is made.